test/testinfra.py

```diff
@@ -264,6 +269,14 @@ def post(self, resource, data, accept=[]):
             raise Exception("GitHub API problem: {0}".format(response['reason'] or status))
         return json.loads(response['data'])
 
+    def delete(self, resource, data, accept=[]):
+        response = self.request("DELETE", resource, json.dumps(data), { "Content-Type": "application/json" })
+        status = response['status']
+        if (status < 200 or status >= 300) and status not in accept:
+            sys.stderr.write("{0}\n{1}\n".format(resource, response['data']))
+            raise Exception("GitHub API problem: {0}".format(response['reason'] or status))
```

Comment:
I guess so. But there's a very common pattern of two callers doing a delete ... in which case both should succeed.